### DISCUSSION TITLE
Compilation errors - new version

### DISCUSSION THREAD

**Discussion by robkwhitehouse** [khoih-prog/ESP32TimerInterrupt#21] [General]
Hi there,

Have just updated my library to the latest version and now the examples do not compile. I have previously been able to compile the examples OK. I haven't tried all the examples but all the examples that I have tried fail because it seems that the interrupt handler (defined in the main sketch) is not in scope inside the library routines (e.g. setFrequency).

I have attached on of the compiler error messages below;

```
In file included from /home/robert/Arduino/libraries/ESP32TimerInterrupt/src/ESP32TimerInterrupt.h:55:0,
                 from /home/robert/Arduino/eCompass/eCompass.ino:59:
/home/robert/Arduino/libraries/ESP32TimerInterrupt/src/ESP32TimerInterrupt.hpp: In member function 'bool ESP32TimerInterrupt::setFrequency(const float&, esp32_timer_callback)':
/home/robert/Arduino/libraries/ESP32TimerInterrupt/src/ESP32TimerInterrupt.hpp:329:100: error: 'timer_isr_callback_add' was not declared in this scope
         timer_isr_callback_add(_timerGroup, _timerIndex, _callback, (void *) (uint32_t) _timerNo, 0);
```
                                                                                                    ^

**Comment by khoih-prog** [comment]
Check your installation as well as ESP32 core version (must be v2.0.2) and this library version (v2.0.0)


![Selection_110](https://user-images.githubusercontent.com/57012152/155759823-daea1718-0d74-4418-9914-7859a3c2f68e.png)
)

**Reply by robkwhitehouse** [reply]
Thankyou - I have now updated to ESP32 board version 2.0.0 and it has solved the problem. Hopefully this board vesrsion will also be OK with the rest of my application
